fdbclient/RESTClient.actor.cpp

```diff
@@ -87,32 +87,36 @@ std::unordered_map<std::string, int> RESTClient::getKnobs() const {
 	return knobs.get();
 }
 
-ACTOR Future<Reference<HTTP::Response>> doRequest_impl(Reference<RESTClient> client,
-                                                       std::string verb,
-                                                       HTTP::Headers headers,
-                                                       RESTUrl url,
-                                                       std::set<unsigned int> successCodes) {
+ACTOR Future<Reference<HTTP::IncomingResponse>> doRequest_impl(Reference<RESTClient> client,
+                                                               std::string verb,
+                                                               HTTP::Headers headers,
+                                                               RESTUrl url,
+                                                               std::set<unsigned int> successCodes) {
+
+	state Reference<HTTP::OutgoingRequest> req = makeReference<HTTP::OutgoingRequest>();
 	state UnsentPacketQueue content;
-	state int contentLen = url.body.size();
+	req->data.content = &content;
+	req->data.contentLen = url.body.size();
+	req->data.headers["Host"] = url.host;
+	req->verb = verb;
+	req->resource = url.resource;
 
 	if (FLOW_KNOBS->REST_LOG_LEVEL >= RESTLogSeverity::VERBOSE) {
 		TraceEvent("RESTDoRequestImpl").detail("Url", url.toString());
 	}
 
 	if (url.body.size() > 0) {
-		PacketWriter pw(content.getWriteBuffer(url.body.size()), nullptr, Unversioned());
+		PacketWriter pw(req->data.content->getWriteBuffer(url.body.size()), nullptr, Unversioned());
 		pw.serializeBytes(url.body);
 	}
 
+	// FIXME: should this be url.host, url.service?
```

Comment:
@sfc-gh-ahusain ?

Reply:
yeah, it should be `url.host, url.service` ; seems like a bug.